### PR TITLE
DP Solution to Subset Count

### PR DESCRIPTION
**Pull request by samratpodder**
### **Describe your change:**

- [x] Add an algorithm?

#### References

Contribution with reference to Issue #3553 

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/SubsetCount.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+
+
+    public static void main(String[] args) {
+        int[] testArray = new int[]{1,2,3,3};
+        System.out.println(new SubsetCount().getCount(testArray,6));//Answer = 3
+        testArray = new int[]{1, 1, 1, 1};
+        System.out.println(new SubsetCount().getCountSO(testArray,1));//Answer = 4
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/dynamicprogramming/SubsetCount.java:77]: Please replace this with JUnit tests

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, could you please add some automated tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

File: src/main/java/com/thealgorithms/dynamicprogramming/SubsetCount.java (hunk 1 above)
Comment (samratpodder) [reply src/main/java/com/thealgorithms/dynamicprogramming/SubsetCount.java]: @siriak Changes have been made. Please take a look at it.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**